app/controllers/patients_controller.rb

```diff
@@ -38,6 +38,10 @@ def show
     @translations = Assessment.new.translations
 
     @history_types = History::HISTORY_TYPES
+
+    # Update viewer cache
+    @viewers = (Rails.cache.read("patient-view-#{@patient.id}")&.to_set || Set.new).delete(current_user.email)
+    Rails.cache.write("patient-view-#{@patient.id}", (@viewers + [current_user.email]).to_a, expires_in: 12.hours)
```

Comment:
One thing to note is that since hitting the back button (going back to the patient show page) will not reload the page sometimes this won't won't add the user to the cache in that case

Reply:
I'm not seeing the on_unload route being hit when navigating away from the page on safari, but it DOES get called when the page is closed on safari

Reply:
`on_unload` isn't perfect, there are a few alternatives - I don't think this will ever be perfect since it's client side, but I'll see what I can do to catch more of those cases.